sdk/openai/azure-ai-openai-assistants/src/test/java/com/azure/ai/openai/assistants/AzureVectorStoreAsyncTests.java

```diff
@@ -0,0 +1,321 @@
+
+            // clean up the created vector store
+            deleteVectorStores(client, vectorStoreId);
+        }, client);
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.ai.openai.assistants.TestUtils#getTestParameters")
+    public void deleteVectorStore(HttpClient httpClient, AssistantsServiceVersion serviceVersion) {
+        client = getAssistantsAsyncClient(httpClient, serviceVersion);
+        deleteVectorStoreAsyncRunner((vectorStoreId) -> {
+            // Delete Vector Store
+            StepVerifier.create(client.deleteVectorStore(vectorStoreId))
+                    .assertNext(deletionStatus -> {
+                        assertTrue(deletionStatus.isDeleted());
+                        assertEquals(deletionStatus.getId(), vectorStoreId);
+                    })
+                    .verifyComplete();
+        }, client);
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.ai.openai.assistants.TestUtils#getTestParameters")
+    public void listVectorStore(HttpClient httpClient, AssistantsServiceVersion serviceVersion) {
+        client = getAssistantsAsyncClient(httpClient, serviceVersion);
+        listVectorStoreAsyncRunner((store1, store2) -> {
+            // List Vector Stores
+            StepVerifier.create(client.listVectorStores())
+                    .assertNext(vectorStores -> {
+                        assertNotNull(vectorStores);
+                        assertFalse(vectorStores.getData().isEmpty());
+                        vectorStores.getData().forEach(vectorStore -> {
+                            assertNotNull(vectorStore.getId());
+                            assertNotNull(vectorStore.getCreatedAt());
+                        });
+                    })
+                    .verifyComplete();
+            // clean up the created vector stores
+            deleteVectorStores(client, store1.getId(), store2.getId());
+        }, client);
+    }
+
+    // Vector Store with Files
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.ai.openai.assistants.TestUtils#getTestParameters")
+    @Disabled("Azure resource won't able to create a vector store with files")
+    public void createVectorStoreFile(HttpClient httpClient, AssistantsServiceVersion serviceVersion) {
+        client = getAssistantsAsyncClient(httpClient, serviceVersion);
+        createVectorStoreWithFileAsyncRunner((storeId, fileId) -> {
+            StepVerifier.create(client.createVectorStoreFile(storeId, fileId))
+                    .assertNext(vectorStoreFile -> {
+                        assertNotNull(vectorStoreFile);
+                        assertNotNull(vectorStoreFile.getId());
+                    })
+                    .verifyComplete();
+            // clean up the created vector store
+            deleteVectorStores(client, storeId);
+            client.deleteFile(fileId).block();
```

Comment:
I was thinking that some of these calls could be pushed to the test runners. Just an idea for one of the many things we can refactor "when we have time"